### ISSUE TITLE
Q: Miss the deploying

### ISSUE THREAD

**Issue by torsten-liermann** [arquillian/arquillian-testcontainers#31]
Hi,

How are the deployments defined in the test deployed, or how does Arquillian receive the corresponding metadata? Or is there still a gap in the implementation here?

Thx.

**Comment by jasondlee** [comment]
The goal, thus far, is not necessarily starting a container to which tests will be deployed, but to start containers that are needed for testing. For example, for WildFly, we're using this to start containers for the OpenTelemetry Collector and Jaeger so that we can test our OpenTelemetry and Micrometer support, which was our primary/initial goal for the library. Being able to deploy tests to these managed  containers, then, isn't what we were aiming for, but I don't see why it couldn't be done. It would, though, require some more integration work, probably much like (probably augmenting or even replacing) the container-specific extensions.

**Comment by torsten-liermann** [comment]
Thanks for the hint. I'm currently looking at the trunk of WildFly. So far, I'm still lacking the knowledge of Arquillian to implement "deploy" and "undeploy." If I get the motivation, I will probably gather some initial ideas from the Arquillian implementations.

**Comment by jamezp** [comment]
I don't think we want to use this to deploy applications. Arquillian itself already does that. The WildFly Arquillian container will handle the deploying/undeploying of applications.

**Comment by torsten-liermann** [comment]
That works for me. I initially categorized the discussions on the topic of Testcontainers the way I use them in the context of Spring Boot and didn’t think much further about it. The application of this implementation by WildFly 34.0.0 is enlightening. Thanks so far.

**Comment by rhusar** [comment]
As @jasondlee explained, the extension starts auxiliary containers needed by tests (e.g., OpenTelemetry Collector, Jaeger), not containers to deploy into.

Presumably, you can still use this to run a container (e.g. WildFly) and then deploy into it as a [remote](https://github.com/wildfly/wildfly-arquillian/tree/main/container-remote) instance? Did we ever test this @jamezp? Should that work?

Nevertheless, I think we can close this.

**Comment by jamezp** [comment]
You can definitely use tools like the [WildFly Plugin Tools](https://github.com/wildfly/wildfly-plugin-tools/) to deploy to a running server from tests. There is a `ServerManager` API and a `DeploymentManager` API.
